pyspecdata/core.py

```diff
@@ -2593,8 +2594,10 @@ def thiserrbarplot(*tebargs,**tebkwargs):
     return retval
 #}}}
 #{{{general functions
-def box_muller(length, return_complex=True):
+def box_muller(length, return_complex=True, seed=None):
     r'''algorithm to generate normally distributed noise'''
+    if seed is not None:
+        pl_seed(seed)
```

Comment:
you shouldn't have to do it this way -- if you seed, the seed should be global (e.g. you can seed in your script), I think

Reply:
Tried it in ipython, and I'm wrong about that
```
In [14]: py_seed(1432)

In [15]: print(a.add_noise(0.5))
[1.07955776, 0.07674935, 1.3582009 , 2.53816454, 4.53411609
 5.04623931, 4.89612688, 6.06584429, 7.21472772, 8.39430309]
                +/-None
        dimlabels=['t']
        axes={`t':[0., 1., 2., 3., 4., 5., 6., 7., 8., 9.]
                        +/-None}


In [16]: py_seed(1432)

In [17]: print(a.add_noise(0.5))
[1.02877369, 0.05866427, 1.18729449, 2.40708142, 4.8559464
 5.16286492, 4.50150268, 5.66294507, 6.66677689, 8.21547053]
                +/-None
        dimlabels=['t']
        axes={`t':[0., 1., 2., 3., 4., 5., 6., 7., 8., 9.]
                        +/-None}
```
(I would expect both to be the same number if seeded in the same way)

Reply:
What I'm saying here is correct ... notice below how seeding the same number results in the same random sequence following the seed, even though two subsequent calls (without a seed between) give back different values:
```
In [18]: a = nddata(r_[0:10.],'t')

In [19]: py_seed(1432)

In [20]: print(a.C.add_noise(2))
[-0.20313629,  0.92765967,  1.31637436,  2.47566753,  5.28732126
  5.46650244,  4.4215032 ,  5.38840311,  5.80819667,  8.28466978]
                +/-None
        dimlabels=['t']
        axes={`t':[0., 1., 2., 3., 4., 5., 6., 7., 8., 9.]
                        +/-None}


In [21]: print(a.C.add_noise(2))
[-0.19759952,  1.7238222 ,  0.60702088,  2.93038397,  3.71326082
  3.65133217,  6.60735107,  7.37541355,  7.94590838,  6.72645224]
                +/-None
        dimlabels=['t']
        axes={`t':[0., 1., 2., 3., 4., 5., 6., 7., 8., 9.]
                        +/-None}


In [22]: py_seed(1432)

In [23]: print(a.C.add_noise(2))
[-0.20313629,  0.92765967,  1.31637436,  2.47566753,  5.28732126
  5.46650244,  4.4215032 ,  5.38840311,  5.80819667,  8.28466978]
                +/-None
        dimlabels=['t']
        axes={`t':[0., 1., 2., 3., 4., 5., 6., 7., 8., 9.]
                        +/-None}

```

Reply:
Ok, that is an interesting example. So to clarify, you're saying what was in place before this is all that is needed, and what I have added with the seed=None and the if statement can all be deleted?

Reply:
I think so.  I'm saying there's no need to reference the seed inside the individual functions -- you can do that just once in th main script.